client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,76 @@
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import React, { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+import { useActions } from './use-actions';
+import { useFields } from './use-fields';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( getPluginActionStatuses );
```

Comment:
Pardon my candour and my lack of context (I haven't touched Calypso since 2017), but `getPluginActionStatuses` really struck me as an odd thing while reviewing this PR. I understand it precedes this PR by a couple of years, but is it really the best way to gather the necessary data for DataViews?

At the very least, there should be some caching, either right here or in `getPluginActionStatuses`. But ideally that function wouldn't [call the same selector 5 times](https://github.com/Automattic/wp-calypso/blob/04e61486e434796d0da8518c874ec3949d4f3c9d/client/my-sites/plugins/plugin-management-v2/utils/get-plugin-action-statuses.ts#L8-L12) to traverse all installed plugin states and produce a single array.

Reply:
This is a good catch. There's no good reason for this wrapper selector, let's just use the underlying selectors.

Reply:
Yeah, that's not ideal. Since this PR is part of a bigger project (pet6gk-1CP-p2), I added your suggestion [to this task](https://github.com/Automattic/dotcom-forge/issues/9362) so we can tackle it as a follow-up since this PR is already big.

Reply:
Actually, [I ended up fixing it here](https://github.com/Automattic/wp-calypso/pull/95561/commits/12c6d2f0d5727162e9a523a2dba45878f89df554).